sql/moz-fx-data-shared-prod/firefox_ios_derived/firefox_ios_clients_v1/query.sql

```diff
@@ -0,0 +1,265 @@
+-- Query first observations for Firefox iOS Clients.
+WITH first_seen AS (
+  SELECT
+    client_id,
+    submission_date,
+    first_run_date,
+    first_seen_date,
+    sample_id,
+    country AS first_reported_country,
+    isp AS first_reported_isp,
+    normalized_channel AS channel,
+    device_manufacturer,
+    device_model,
+    normalized_os_version AS os_version,
+    app_display_version AS app_version
+  FROM
+    firefox_ios.baseline_clients_first_seen
+  WHERE
+    submission_date = @submission_date
+    AND client_id IS NOT NULL
+    -- TODO: is this meant to be restricted to release channel only?
+    AND normalized_channel = 'release'
+),
+-- Find the most recent activation record per client_id.
+activations AS (
+  SELECT
+    client_id,
+    ARRAY_AGG(activated ORDER BY submission_date DESC)[SAFE_OFFSET(0)] > 0 AS activated
+  FROM
+    firefox_ios.new_profile_activation
+  WHERE
+    submission_date = @submission_date
+  GROUP BY
+    client_id
+),
+-- Find earliest data per client from the first_session ping.
+first_session_ping AS (
+  SELECT
+    client_id,
+    MIN(sample_id) AS sample_id,
+    DATETIME(MIN(submission_timestamp)) AS min_submission_datetime,
+    DATE(MIN(SAFE.PARSE_DATETIME('%F', SUBSTR(first_run_date, 1, 10)))) AS first_run_date,
+    ARRAY_AGG(
+      IF(
+        -- we should probably do the nullif prior to doing this is not null check
+        adjust_ad_group IS NOT NULL
+        OR adjust_campaign IS NOT NULL
+        OR adjust_creative IS NOT NULL
+        OR adjust_network IS NOT NULL,
+        STRUCT(
+          submission_timestamp,
+          adjust_ad_group,
+          adjust_campaign,
+          adjust_creative,
+          adjust_network
+        ),
+        NULL
+      ) IGNORE NULLS
+      ORDER BY
+        submission_timestamp
+      LIMIT
+        1
+    )[SAFE_OFFSET(0)] AS adjust_info,
+  FROM
+    (
+      SELECT
+        client_info.client_id,
+        sample_id,
+        submission_timestamp,
+        client_info.first_run_date,
+        NULLIF(metrics.string.adjust_ad_group, "") AS adjust_ad_group,
+        NULLIF(metrics.string.adjust_campaign, "") AS adjust_campaign,
+        NULLIF(metrics.string.adjust_creative, "") AS adjust_creative,
+        NULLIF(metrics.string.adjust_network, "") AS adjust_network,
+      FROM
+        firefox_ios.first_session
+      WHERE
+        DATE(submission_timestamp) = @submission_date
+        AND client_info.client_id IS NOT NULL
+        AND ping_info.seq = 0 -- Pings are sent in sequence, this guarantees that the first one is returned.
```

Comment:
We should drop this filter. Take the first attribution we get, even if it comes from a `first_session` ping with `seq = 1`.